### PR TITLE
leo-editor: 5.5 -> 5.6, add docutils and use pyqt56

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Update leo to new version, recently released.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
pyqt56 is introduced and used to fix themes, I tried it on a whim since after seeing it mentioned somewhere.  It's our normal pyqt built against 5.6 instead of 5.9.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -257,6 +257,10 @@ in {
     pythonPackages = self;
   };
 
+  pyqt56 = pkgs.libsForQt56.callPackage ../development/python-modules/pyqt/5.x.nix {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Keep this private to leo-editor. We shouldn't have multiple versions of a package in `python-packages.nix`

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (dtzWill) [reply pkgs/top-level/python-packages.nix]: Okay, sure thing.  Sorry about that (pushed new commit with requested change).

**Comment by dtzWill** [comment]
rebased to latest master, ping!

File: pkgs/applications/editors/leo-editor/default.nix [hunk 2]
```diff
@@ -1,20 +1,20 @@
-{ stdenv, python3Packages, fetchFromGitHub, makeWrapper, makeDesktopItem }:
+{ stdenv, python3Packages, pyqt56, fetchFromGitHub, makeWrapper, makeDesktopItem }:
```
Comment (FRidh) [inline pkgs/applications/editors/leo-editor/default.nix]: Overriding the Python package set here, replacing the `pyqt5` attribute would be the preferred solution.

Comment (FRidh) [reply pkgs/applications/editors/leo-editor/default.nix]: mitmproxy is a good example

**Comment by dtzWill** [comment]
Sorry for the delay--pushed new commit, quite a bit simpler!

Turns out there's already a `pyqt56` in the python package set, so just using that instead.

**Comment by joachifm** [comment]
I believe the review comments were addressed.  Thank you

**Comment by joachifm** [comment]
Ah, turns out pyqt56 ended up being removed at 964da1b91a115c7a3e4dee64420d8b7dd894b408. My bad.

**Comment by dtzWill** [comment]
Erm, I'll open new one?

**Pull request merged**